azurerm/data_source_healthcare_service.go

```diff
@@ -0,0 +1,165 @@
+					Schema: map[string]*schema.Schema{
+						"allowed_origins": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validate.NoEmptyStrings,
+							},
+						},
+						"allowed_headers": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validate.NoEmptyStrings,
+							},
+						},
+						"allowed_methods": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validate.NoEmptyStrings,
+							},
+						},
+						"max_age_in_seconds": {
+							Type:     schema.TypeInt,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validate.NoEmptyStrings,
+							},
+						},
+						"allow_credentials": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func dataSourceArmHealthcareServiceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Healthcare.HealthcareServiceClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[WARN] Healthcare Service %q was not found (Resource Group %q)", name, resourceGroup)
+			d.SetId("")
+			return fmt.Errorf("HealthCare Service %q was not found in Resource Group %q", name, resourceGroup)
+		}
+		return fmt.Errorf("Error making Read request on Azure Healthcare Service %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+	if kind := resp.Kind; string(kind) != "" {
+		d.Set("kind", kind)
+	}
+
+	if properties := resp.Properties; properties != nil {
+		if accessPolicies := properties.AccessPolicies; accessPolicies != nil {
+			d.Set("access_policy_object_ids", flattenHealthcareAccessPolicies(accessPolicies))
+		}
+
+		if config := properties.CosmosDbConfiguration; config != nil {
+			d.Set("cosmosdb_throughput", config.OfferThroughput)
+		}
+
+		if authConfig := properties.AuthenticationConfiguration; authConfig != nil {
+			if err := d.Set("authentication_configuration", flattenHealthcareAuthConfig(authConfig)); err != nil {
+				return fmt.Errorf("Error setting `authentication_configuration`: %+v", flattenHealthcareAuthConfig(authConfig))
+			}
+		}
+
+		if corsConfig := properties.CorsConfiguration; corsConfig != nil {
+			if err := d.Set("cors_configuration", flattenHealthcareCorsConfig(corsConfig)); err != nil {
+				return fmt.Errorf("Error setting `cors_configuration`: %+v", flattenHealthcareCorsConfig(corsConfig))
+			}
+		}
+	}
+
+	flattenAndSetTags(d, resp.Tags)
```

Comment:
Could we use the new
```suggestion
	return tags.FlattenAndSet(d, resp.Tags)
```